frontend_lib/src/component/Timeline/Timeline.jsx

```diff
@@ -31,28 +31,40 @@ export class Timeline extends React.Component {
 
   componentDidMount () {
     this.timelineContainerScrollHeight = this.timelineContainer.scrollHeight
-    this.scrollToBottom()
+    if (window.innerWidth < 1200) return
+    this.timelineBottom.scrollIntoView({ behavior: 'instant' })
   }
 
-  componentDidUpdate () {
-    if (window.innerWidth < 1200) return
-    this.props.shouldScrollToBottom && this.scrollToBottom()
-    if (this.timelineContainerScrollHeight !== this.timelineContainer.scrollHeight) {
-      this.timelineContainerScrollHeight = this.timelineContainer.scrollHeight
-    }
+  componentDidUpdate (prevProps) {
+    if (this.props.shouldScrollToBottom && this.props.timelineData && prevProps.timelineData) this.scrollToBottom(prevProps.timelineData)
+    this.timelineContainerScrollHeight = this.timelineContainer.scrollHeight
   }
 
-  scrollToBottom = () => {
+  scrollToBottom = (prevTimeline) => {
     const { props } = this
-    // GM - INFO - 2020-06-30 - Check if the timeline scroll is at the bottom or if the new comment was created by the current session tokenId
+
+    if (props.timelineData.length === 0) return
+
+    const isNewContent = prevTimeline.length > 0
+      ? this.getTimelineContentId(prevTimeline[prevTimeline.length - 1]) !== this.getTimelineContentId(props.timelineData[props.timelineData.length - 1])
+      : false
+    const isScrollAtTheBottom = this.timelineContainer.scrollTop + this.timelineContainer.clientHeight === this.timelineContainerScrollHeight
+
+    // GM - INFO - 2020-06-30 - When width >= 1200: Check if the timeline scroll is at the bottom or if the new item was created by the current session tokenId or if the content_id has changed
+    // When width >= 1200: Check the if the new comment was created by the current session tokenId
     if (
-      this.timelineContainer.scrollTop + this.timelineContainer.clientHeight === this.timelineContainerScrollHeight ||
-      (props.isLastTimelineItemCurrentToken && props.newComment === '')
+      (window.innerWidth >= 1200 && (isNewContent || isScrollAtTheBottom || (props.isLastTimelineItemCurrentToken && props.newComment === ''))) ||
+      (window.innerWidth < 1200 && props.isLastTimelineItemCurrentToken && props.newComment === '' && props.timelineData[props.timelineData.length - 1].content_type === CONTENT_TYPE.COMMENT)
     ) {
-      this.timelineBottom.scrollIntoView({ behavior: 'instant' })
+      const behavior = isScrollAtTheBottom && props.isLastTimelineItemCurrentToken ? 'smooth' : 'instant'
+      this.timelineBottom.scrollIntoView({ behavior })
     }
   }
```

Comment:
Your lines are too long.
Github displays around 150 columns which is already pretty big but here, your code can't even fit on github pages.
Usually, in javascript, IDE advice lines length at 120 columns. We don't have a strict rule on that but some of your lines are near 200 columns.
For example, the PEP8 line length coding rule in python is 80 columns.

It's always painful to have to scroll horizontally, especially on desktop (without trackpad).

Reply:
Also your last test is still a bit too complicated.
You did well with the variables isNewContent and isScrollAtBottom, try to rename some more conditions

It'll also shorten the lines

Reply:
for example, I see that `props.isLastTimelineItemCurrentToken && props.newComment === '')` is on both innerWidth >= 1200 and < 1200.
Couldn't you have an early return if this condition isn't met ?

Reply:
We couldn't have an early return if this condition isn't met because with `innerWidth >= 120` it is in a OR condition

Reply:
indeed